labgrid/driver/power/siglent.py

```diff
@@ -0,0 +1,21 @@
+""" tested with Siglent SPD3303X-E, and should be compatible with all SPD3000X series modules"""
+
+import vxi11
+
+
+def power_set(host, index, value):
+    index = int(index)
+    assert 1 <= index <= 2
+    value = "ON" if value else "OFF"
+    psu = vxi11.Instrument(host)
```

Comment:
Is this using a fixed TCP port? If so, you should set `PORT` in the backend module, as then labgrid can do SSH port forwarding. (see netio_kshell.py)

Reply:
It is a bit more complicated than that.  VXI-11 protocol specifes both a TCP and a UDP port.

Reply:
Hmm, can it work with either? How would it select TCP vs. UDP?

Reply:
I don't know, I just used it, and it worked :)

Reply:
Actually, looking at the python-vxi11 code, it seems like UDP is only used for VXI-11 device discovery, which doesn't seem to work with the Siglent device I have here anyway.

But, the TCP communication uses port 111 (RPC/portmap), which returns another port, which is then used for the remaining communication.  In my case it was 9009, but I don't know if that is guaranteed to be a fixed port number.

Reply:
Oh, OK. Why do people use portmapper...

The leave out the PORT entry and we'll have to figure out forwarding later, if needed.